library/blob.c

```diff
@@ -1057,6 +1053,82 @@ static int eblob_blob_iterator(struct eblob_iterate_priv *iter_priv)
 	return err;
 }
 
+
+/**
+ * eblob_disk_control_cmp_by_position() - compares dc's by position in a blob file
+ */
+static int eblob_disk_control_cmp_by_position(const void *lhs, const void *rhs) {
+	const struct eblob_disk_control *left = (const struct eblob_disk_control *)lhs;
+	const struct eblob_disk_control *right = (const struct eblob_disk_control *)rhs;
+	if (left->position < right->position) {
+		return -1;
+	} else if (left->position > right->position) {
+		return 1;
+	} else {
+		return 0;
+	}
+}
+
+/**
+ * eblob_blob_iterate_by_pos() - read whole index, sort it by position and run iteration over it.
+ */
+static int eblob_blob_iterate_by_pos(struct eblob_iterate_priv *iter_priv)
+{
+	struct eblob_iterate_control *ctl;
+	struct eblob_base_ctl *bctl;
+
+	struct eblob_iterate_local loc;
+	int err = 0;
+
+	static const int hdr_size = sizeof(struct eblob_disk_control);
+
+	if (iter_priv == NULL) {
+		err = -EINVAL;
+		return err;
+	}
+
+	ctl = iter_priv->ctl;
+	bctl = ctl->base;
+
+	memset(&loc, 0, sizeof(loc));
+
+	loc.iter_priv = iter_priv;
+	loc.index_offset = 0;
+	loc.pos = 0;
+	loc.num = bctl->index_ctl.size / hdr_size;
+
+	loc.dc = malloc(bctl->index_ctl.size);
+	if (loc.dc == NULL) {
```

Comment:
minor: maybe we could used same rules as we agreed on elliptics not to check for ENOMEM?

Reply:
done